1Password/docusaurus-extensions
# [DEV] Version 1.0.0 of @1password/docusaurus-plugin-stored-data and @1password/jest-docusaurus (Breaking Change)

## Summary

Bump the versions of `@1password/docusaurus-plugin-stored-data` and `@1password/jest-docusaurus` from `0.1.0` to `1.0.0`. _This MR provides changes that shift support from Docusaurus 2.x to 3.x (breaking changes)_. We have specifically updated our dependencies to support Docusaurus 3.2.1.

Within this MR, updates to prerequisite dependencies have been made. Information on these requirements can be found via: [Docusaurus' preparation guidelines for Docusaurus 3](https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3) as well as the [Official Migration Guide](https://docusaurus.io/docs/migration/v3).

## Related Issues and MRs

Closing #65 

## What changed

### Prerequisites for Docusaurus 3
- Upgrade `react`, `react-dom`, and `@types/react` to 18.2.0
- Upgrade `typescript` to 5.2.2
- Upgrade `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` to 6.5.0
- Upgrade `ts-jest` to 29.1.0
- Remove `@tsconfig/docusaurus` and install/extend `@docusaurus/tsconfig`
- Switch from yarn 2+ [plug and play to normal node_modules structure](https://yarnpkg.com/features/pnp#when-migrating-an-existing-project)
- API changes from TypeScript updates
- Upgrade `jest` and `jest-environment-jsdom` to 29.7.0
- Upgrade `@types/jest` to 29.5.12
- Upgrade `@jest/types` to 29.6.3
- Upgrade `eslint-plugin-jest` to 28.2.0
- Upgrade `@testing-library/jest-dom` to 6.4.2
- Support `Node.js` 18.19.1 

### Docusaurus 3
- Upgrade `@mdx-js/react` to 3.0.1
- Upgrade `@docusaurus/core` and adjacent packages:`@docusaurus/types`, `@docusaurus/module-type-aliases`, `@docusaurus/theme-classic`, `@docusaurus/preset-classic`, `@docusaurus/utils-validation`, and `@docusaurus/plugin-content-docs`, and `@docusaurus/eslint-plugin` to ^3.2.1
- Resolve breaking changes from Docusaurus 3+ (imports + api changes)

## Steps for reviewer

1. Pull changes and install
```
git checkout brian/docusaurus-3-compatible
yarn install
```
2. Run both plugins locally
```
yarn dev docusaurus-plugin-stored-data
```
```
yarn dev jest-docusaurus
```
3. Verify tests are passing
```
yarn test
```
4. Ensure there was no diff
```
yarn syncpack
```
5. Build all of the packages and demo website
```
yarn build:all
```

## Additional resources

- https://docusaurus.io/docs/migration/v3
- https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3
